app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestActivity.java

```diff
@@ -0,0 +1,162 @@
+    Toolbar toolbar;
+    @BindView(R.id.fragment_container)
+    ViewPager viewPager;
+    private SlidingTabAdapter tabAdapter;
+
+    public static Intent createIntent(Context context, UUID creatorId) {
+        return new Intent(context, NewContestActivity.class).putExtra(KEY_CREATOR_ID, creatorId);
+    }
+
+    @NonNull
+    @Override
+    public NewContestPresenter createPresenter(PresenterConfiguration configuration) {
+        return new NewContestPresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_viewpager_container;
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ButterKnife.bind(this);
+        setupToolbar();
+        setupViewPager(viewPager);
+        presenter.initializeContestBuilder(getIntent());
+    }
+
+    public NewContestContract.Presenter getPresenter() {
+        return presenter;
+    }
+
+    private void setupToolbar() {
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setTitle(R.string.new_contest);
+        actionBar.setHomeAsUpIndicator(android.R.drawable.ic_media_rew);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            setupStatusBarColor();
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private void setupStatusBarColor() {
+        Window window = getWindow();
+        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        window.setStatusBarColor(getResources().getColor(R.color.colorPrimaryDark));
+    }
+
+    public void setupViewPager(ViewPager viewPager) {
+        tabAdapter = new SlidingTabAdapter(this);
+        tabAdapter.addFragment(new NameContestFragment());
+        tabAdapter.addFragment(new DescribeContestFragment());
+        tabAdapter.addFragment(new CategoriesListFragment());
+        viewPager.setAdapter(tabAdapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        menu.clear();
+        getMenuInflater().inflate(R.menu.menu_new_contest, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                presenter.onBackButtonClicked();
+                break;
+            case R.id.action_next:
+                presenter.onNextButtonClicked();
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public int getNumberOfScreens() {
+        return NUMBER_OF_SCREENS;
+    }
+
+    public void acceptContestName(String contestName) {
+        presenter.setContestName(contestName);
```

Comment:
I was thinking onContestNameEntered() so that it is clear that while it's not a callback, its happening in response to a user input.. whatcha think?